moto/core/models.py

```diff
@@ -116,9 +120,9 @@ def stop(self, remove_data: bool = True) -> None:
             self.disable_patching()  # type: ignore[attr-defined]
 
     def decorate_callable(
-        self, func: Callable[..., CALLABLE_RETURN], reset: bool, remove_data: bool
-    ) -> Callable[..., CALLABLE_RETURN]:
-        def wrapper(*args: Any, **kwargs: Any) -> CALLABLE_RETURN:
+        self, func: Callable[..., "BaseMockAWS"], reset: bool, remove_data: bool
+    ) -> Callable[..., "BaseMockAWS"]:
+        def wrapper(*args: Any, **kwargs: Any) -> "BaseMockAWS":
```

Comment:
`func` is the user’s function being decorated. There’s no reason to expect it to return `BaseMockAWS`. All we can say is that our wrapper will return the same type that the user’s function returned, whatever it is. The type variable `CALLABLE_RETURN` was more correct.